service/frontend/fx.go

```diff
@@ -343,22 +343,32 @@ func TelemetryInterceptorProvider(
 	)
 }
 
+func getRateFnWithMetrics(rateFn quotas.RateFn, handler metrics.Handler) quotas.RateFn {
+	return func() float64 {
+		rate := rateFn()
+		metrics.HostRPSLimit.With(handler).Record(rate)
```

Comment:
It'd be great if we could only emit this when the rate changes rather than on every call to the rate limiter, but that may be more work than it's worth until dynamic config supports that sort of behavior (like we discussed on the DLQ regex pr)

Reply:
Yeah, I thought about implementing such a logic in this function itself. Cache the previous value to a variable and compare it with the new value before emitting the metric. 

But it might not give us any performance improvement. When we emit a metric, it just updates a value in memory. Caching the previous value and comparing with new value is more work than that.

But if there is synchronization when updating metrics, we could get some performance boost by caching previous value.